sim/battle-queue.ts

```diff
@@ -18,7 +18,7 @@ import type {Battle} from './battle';
 /** A move action */
 export interface MoveAction {
 	/** action type */
-	choice: 'move' | 'beforeTurnMove';
+	choice: 'move' | 'beforeTurnMove' | 'priorityChargeMove';
 	order: 3 | 5 | 200 | 201 | 199;
```

Comment:
This list should probably have your new order here.

Personally I'm not a fan of the code duplication that Gen 4 requires. Is it possible keep the `beforeTurnMove` action but just change its choice and/or order in Gen 4?